test/stateful_dataloader/test_state_dict.py

```diff
@@ -1529,7 +1530,8 @@ def state_dict(self):
         return {"iter_calls": self.iter_calls, "items": deepcopy(self.items)}
 
     def load_state_dict(self, state_dict):
-        self.iter_calls = state_dict["iter_calls"]
+        # sequence of calls for this : iter is called first and then load_state_dict is called and thus we don't want state to override calls to iter that has happened before load_state_dict was called
+        self.iter_calls += state_dict["iter_calls"]
```

Comment:
This makes load_state_dict non-idempotent, so eg if load_state_dict is called multiple times it will result in incorrect behaviour

Reply:
@andrewkho Updated the unit test to check specifically the iter_calls alone and not restore it value which is unnecessary